File: S03/week02/sejoong/MinimumSizeSubarraySum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        List<Integer> sortedNums = Arrays.stream(nums)
            .boxed()
            .sorted(Comparator.reverseOrder())
            .toList();

        int sum = 0;
        int subArrayLen = 0;
        while (target > sum && subArrayLen < sortedNums.size()) {
            sum += sortedNums.get(subArrayLen);
            subArrayLen++;
        }
        
        if (target > sum) {
            return 0;
        }

        return subArrayLen;
    }
}

File: S03/week02/seokjoon/MinimumSizeSubarraySum.kt
//286ms , 48.93MB
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        var l = 0
        var r = 0
        var minSize = nums.size + 1
        var curSum = 0
        while(r < nums.size && l <= r){
            curSum += nums[r++] 
            while(curSum >= target && l < r){
                minSize = minOf(minSize, r - l)
                curSum -= nums[l++]
            }
        }
        return if(minSize == nums.size + 1) 0 else minSize
    }
}
